### PR TITLE
Implement progress bar for when the wallet is downloading

### LINKED ISSUE #1303: Improve "preparing sheikah" view
Currently, Sheikah doesn't give feedback to the user about what is happening during "We are preparing Sheikah" view. In order to improve this, Sheikah could show a progress bar with the different steps:
- Download the wallet
- Run the wallet
- Run Sheikah

### PR DESCRIPTION
**Pull request by gabaldon**
This pr creates a progress bar for when the wallet is downloading.
close #1303

### CODE REVIEW

**Comment by aesedepece** [comment]
Why does this say "21 files changed"? :open_mouth:

File: src/main.js [hunk 1]
```diff
@@ -12,7 +12,24 @@ ipcRenderer.on('shutdown', async () => {
...
+})
+ipcRenderer.on('load-url', async (event, message) => {
+  store.commit('setMessage', { message: message })
+})
+ipcRenderer.on('else', async () => {
+  store.commit('setMessage', { message: 'else' })
+})
+ipcRenderer.on('mal', async () => {
+  store.commit('setMessage', { message: 'MAAAAAAALLLLLLL' })
+})
```
Comment (Tommytrg) [inline src/main.js:32]: :rofl:

File: src/background.js [hunk 2]
```diff
@@ -239,16 +240,21 @@ async function downloadWalletRelease(releaseUrl, version) {
...
     axios
       .get(releaseUrl, { responseType: 'stream' })
       .then(async response => {
         const file = `witnet-release-${arch}-${platform}.tar.gz`
+        const str = progress({
+          length: response.headers['content-length'],
+          time: 100 /* ms */
+        })
+        str.on('progress', function(progress) {
+          win.webContents.send('progress', progress)
```
Comment (Tommytrg) [inline src/background.js:253]:
```suggestion
          win.webContents.send('progress', progress.percentage)
```

File: src/background.js [hunk 3]
```diff
@@ -349,16 +350,15 @@ function main() {
...
       if (!isLastestVersion) {
-        console.info(
-          'There is a newer version. Downloading latest wallet release...',
-        )
+        win.webContents.send('downloading')
+        await sleep(5000)
         await downloadWalletRelease(releaseUrl, latestReleaseVersion)
       } else {
+        win.webContents.send('up-to-date')
+        await sleep(5000)
```
Comment (Tommytrg) [inline src/background.js:359]:
```suggestion
        await sleep(2500)
```

File: src/background.js [hunk 4]
```diff
@@ -349,16 +350,15 @@ function main() {
         existVersionFile &&
         latestReleaseVersion ===
           fs.readFileSync(path.join(SHEIKAH_PATH, VERSION_FILE_NAME)).toString()
-
       if (!isLastestVersion) {
-        console.info(
-          'There is a newer version. Downloading latest wallet release...',
-        )
+        win.webContents.send('downloading')
+        await sleep(5000)
```
Comment (Tommytrg) [inline src/background.js:355]:
```suggestion
        await sleep(2500)
```

File: src/background.js [hunk 5]
```diff
@@ -372,6 +372,8 @@ function main() {
 // Run Witnet wallet and load "ready" url
 async function runWallet() {
   console.info('Running wallet...')
+  win.webContents.send('running')
+  await sleep(8000)
```
Comment (Tommytrg) [inline src/background.js:376]:
```suggestion
  await sleep(3000)
```

File: src/background.js [hunk 6]
```diff
@@ -267,6 +273,7 @@ async function downloadWalletRelease(releaseUrl, version) {
 
         // Remove the compressed file
         fs.unlinkSync(file)
+        await sleep(8000)
```
Comment (Tommytrg) [inline src/background.js:276]:
```suggestion
        await sleep(3000)
```

File: src/main.js [hunk 7]
```diff
@@ -12,7 +12,22 @@ ipcRenderer.on('shutdown', async () => {
...
+ipcRenderer.on('loaded', async () => {
+  store.commit('setMessage', { message: 'loaded' })
+})
+ipcRenderer.on('up-to-date', async () => {
+  store.commit('setMessage', { message: 'wallet up to date' })
+})
+ipcRenderer.on('progress', async (event, progress) => {
+  console.log('.........', progress)
+  store.commit('setProgress', { progress: progress })
+})
```
Comment (Tommytrg) [inline src/main.js:30]: We could move all the ipc handlers to its own file

File: src/views/Setup.vue [hunk 8]
```diff
@@ -29,33 +107,59 @@ export default {
...
+          color: $alpha-purple;
+          font-size: 50px;
+          position: absolute;
+          z-index: 100;
+        }
+      }
 
-      .link {
+      .progress-title {
+        // max-width: 250px;
```
Comment (Tommytrg) [inline src/views/Setup.vue:135]: Could you remove this comment?

File: src/views/Setup.vue [hunk 9]
```diff
@@ -1,20 +1,98 @@
...
+              class="spinner"
+            />
+          </p>
+        </div>
       </div>
-    </Card>
+      <el-progress
+        :text-inside="true"
+        :stroke-width="20"
+        :percentage="percentage"
```
Comment (Tommytrg) [inline src/views/Setup.vue:42]:
```suggestion
        :percentage="Math.round(percentage)"
```

File: src/views/Setup.vue [hunk 10]
```diff
@@ -1,20 +1,98 @@
...
+        return 60
+      } else if (this.setupMessage === 'Running wallet') {
+        return 66
+      } else if (this.setupMessage === 'loaded') {
+        return 100
+      } else {
+        return 0
+      }
+    },
+  },
```
Comment (Tommytrg) [inline src/views/Setup.vue:76]:
```suggestion
    percentage() {
      if (this.setupMessage === 'Downloading wallet') {
        return Math.round(this.progress * 0.8)
      } else if (this.setupMessage === 'wallet up to date') {
        return 80
      } else if (this.setupMessage === 'Running wallet') {
        return 90
      } else if (this.setupMessage === 'loaded') {
        return 100
      } else {
        return 0
      }
    },
  },
```

File: src/views/Setup.vue [hunk 11]
```diff
@@ -1,20 +1,98 @@
...
+        return 'Downloading wallet'
+      } else if (percentage <= 60 && percentage >= 33) {
+        return 'The wallet is up to date'
+      } else if (percentage <= 70) {
+        return 'Running wallet'
+      } else if (percentage > 70) {
+        return 'We are preparing Sheikah'
+      }
+    },
+  },
```
Comment (Tommytrg) [inline src/views/Setup.vue:95]: What do you think about this?

```suggestion
    format(percentage) {
      if (percentage <= 80) {
        return 'Downloading wallet'
      } else if (percentage <= 90) {
        return 'Running wallet'
      } else if (percentage > 90) {
        return 'We are preparing Sheikah'
      }
    },
  },
```

File: src/views/Setup.vue [hunk 12]
```diff
@@ -1,20 +1,98 @@
...
+  computed: {
+    ...mapState({
+      setupMessage: state => state.uiInteractions.setupMessage,
+      client: state => state.wallet.api.client,
+      progress: state => state.uiInteractions.setupProgress,
+    }),
+    percentage() {
+      if (this.setupMessage === 'Downloading wallet') {
+        return this.progress / 4
```
Comment (Tommytrg) [inline src/views/Setup.vue:65]: Avoid show float numbers
```suggestion
        return Math.round(this.progress / 4)
```

**Review by Tommytrg — APPROVED** [review]

**Pull request merged**